File: server/src/test/java/de/zalando/zally/rules/DisabledAvoidJavascriptKeywordsRuleTest.java
package de.zalando.zally.rules;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.stream.Collectors;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest
public class DisabledAvoidJavascriptKeywordsRuleTest {

    @Autowired
    private RulesValidator rulesValidator;

    @Test
    public void ruleShouldNotBePartOfRulesValidator() {
        assertThat(rulesValidator.getRules().stream()
                .map(r -> r.getClass()).collect(Collectors.<Class>toList()))
                .doesNotContain(AvoidJavascriptKeywordsRule.class);
    }
}

File: server/src/test/java/de/zalando/zally/rule/DisabledAvoidJavascriptKeywordsRuleTest.kt
package de.zalando.zally.rule

import de.zalando.zally.getResourceContent
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner

@RunWith(SpringRunner::class)
@SpringBootTest
@ActiveProfiles("test")
class DisabledAvoidJavascriptKeywordsRuleTest {

    @Autowired
    private val rulesValidator: SwaggerRulesValidator? = null

    @Test
    fun ruleShouldNotBePartOfRulesValidator() {
        val violatedRulesClasses = rulesValidator!!.validate(getResourceContent("avoidJavascriptInvalid.json"))
                .map { (rule) -> rule.javaClass.name }
        assertThat(violatedRulesClasses).doesNotContain(AvoidJavascriptKeywordsRule::class.java.name)
    }
}
